c7n/resources/fsx.py

```diff
@@ -329,3 +333,84 @@ class KmsFilter(KmsRelatedFilter):
 class KmsFilterFsxBackup(KmsRelatedFilter):
 
     RelatedIdsExpression = 'KmsKeyId'
+
+
+@FSx.filter_registry.register('consecutive-backups')
+class ConsecutiveBackups(Filter):
+    """Returns consecutive daily FSx backups, which are equal to/or greater than n days.
+    :Example:
+    .. code-block:: yaml
+            policies:
+              - name: fsx-daily-backup-count
+                resource: fsx
+                filters:
+                  - type: consecutive-backups
+                    days: 5
+                actions:
+                  - notify
+    """
+    schema = type_schema('consecutive-backups', days={'type': 'number',
+                                                      'minimum': 1},
+                         required=['days'])
+    permissions = ('fsx:DescribeBackups', 'fsx:DescribeVolumes',)
+    annotation = 'c7n:FSxBackups'
+
+    def process_resource_set(self, client, resources):
+        ontap_fids, nonontap_fids = [], []
+        for r in resources:
+            if r['FileSystemType'] == 'ONTAP':
+                ontap_fids.append(r['FileSystemId'])
+            else:
+                nonontap_fids.append(r['FileSystemId'])
+        vpaginator = client.get_paginator('describe_volumes')
+        bpaginator = client.get_paginator('describe_backups')
+
+        vpaginator.PAGE_ITERATOR_CLS = RetryPageIterator
+        ontap_volumes = vpaginator.paginate(Filters=[
+            {
+                'Name': 'file-system-id',
+                'Values': ontap_fids,
```

Comment:
is there a max values for filesystem ids here when filtering?

Reply:
No max values for filesystem ids, as the filesystems ids are taken from resource output.